demo-apps/cf-android/app/src/main/java/org/eclipse/californium/examples/ConfigureDtls.java

```diff
@@ -0,0 +1,86 @@
+/*******************************************************************************
+ * Copyright (c) 2015 Institute for Pervasive Computing, ETH Zurich and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ *
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ *
+ * Contributors:
+ *    Vikram  - Initial creation
+ ******************************************************************************/
+package org.eclipse.californium.examples;
+
+import android.content.Context;
+
+import org.eclipse.californium.scandium.config.DtlsConnectorConfig;
+import org.eclipse.californium.scandium.dtls.pskstore.StaticPskStore;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.security.GeneralSecurityException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivateKey;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
+
+public class ConfigureDtls {
+
+    private static final boolean PSK_MODE = true;
+    private static final boolean CERTIFICATE_MODE = false;
+    private static final boolean RPK_MODE = false;
+    public static final String PSK_IDENTITY = "password";
+    public static final byte[] PSK_SECRET = "sesame".getBytes();
+    private static final String TRUST_NAME = "root";
+    private static final char[] TRUST_STORE_PASSWORD = "rootPass".toCharArray();
+    private static final char[] KEY_STORE_PASSWORD = "endPass".toCharArray();
+
+    public static void loadCredentials(Context context, DtlsConnectorConfig.Builder dtlsConfig, String alias){
+        try {
+
+            // load key store
```

Comment:
Any reason not to use the `SslContextUtil`?

Reply:
Initially I tired to use android URI to locate the path of the resource folder (truststore.bks and keystore.bks locations) and call .toString() method on it to get the classpath provided to SslContextUtil but it did not work.

Reply:
Try

```
    private static final char[] KEY_STORE_PASSWORD = "endPass".toCharArray();
    private static final String KEY_STORE_LOCATION = "certs/keyStore.p12";
    private static final char[] TRUST_STORE_PASSWORD = "rootPass".toCharArray();
    private static final String TRUST_STORE_LOCATION = "certs/trustStore.p12";
    private static final String CLIENT_NAME = "client";
    private static final String CLIENT_TRUST = null; // all certificates in trust store

    .... load certificates method ...

        SslContextUtil.Credentials clientCredentials = null;
        Certificate[] trustedCertificates = null;
        try {
            clientCredentials = SslContextUtil.loadCredentials(
                    SslContextUtil.CLASSPATH_SCHEME + KEY_STORE_LOCATION, CLIENT_NAME, KEY_STORE_PASSWORD,
                    KEY_STORE_PASSWORD);
            trustedCertificates = SslContextUtil.loadTrustedCertificates(
                    SslContextUtil.CLASSPATH_SCHEME + TRUST_STORE_LOCATION, CLIENT_TRUST, TRUST_STORE_PASSWORD);
        } catch (GeneralSecurityException e) {
            e.printStackTrace();
        } catch (IOException e) {
            e.printStackTrace();
        }
```